fdbserver/MasterProxyServer.actor.cpp

```diff
@@ -1253,6 +1255,7 @@ ACTOR Future<Void> commitBatch(
 			std::min(SERVER_KNOBS->COMMIT_TRANSACTION_BATCH_INTERVAL_MAX, 
 				target_latency * SERVER_KNOBS->COMMIT_TRANSACTION_BATCH_INTERVAL_SMOOTHER_ALPHA + self->commitBatchInterval * (1-SERVER_KNOBS->COMMIT_TRANSACTION_BATCH_INTERVAL_SMOOTHER_ALPHA)));
 
+	self->stats.commitBatchInterval = self->commitBatchInterval;
```

Comment:
I am not sure if the raw number is very useful. It seems the interval is updated/calculated after each commit batch. Every second there should be dozens or more batches. So I guess a histogram is a better choice.

Another metric, i.e., queue length, is more indicative of proxy busyness, which should be better than this delay.

Reply:
> So I guess a histogram is a better choice

IMO, point value logging is enough to tell whether the proxy is about to saturate. I.e. if the value logged has been high for a certain period of time, it's safe to say the proxy is quite overloaded. For this purpose, I don't see the advantage of having 90%, 99% or 99.99% distribution of history values compared to point value.

> Another metric, i.e., queue length, is more indicative of proxy busyness, which should be better than this delay.

Which queue are you talking about?

Reply:
Point logging is once every 5s, the sampling rate is quite low (maybe < 1%). I am not sure the chance of the value happen to be low while proxy is very busy. Histogram can eliminate such uncertainty.

By queue length, I mean queued transactions. This can be obtained by computing the difference of admitted transactions, and the resolving transactions. For instance, we know in some cases the queuing time can be as high as 80s, which means there must be a lot of transactions in the queue. Another queue length is those at logging phase, which can be useful as well.

Reply:
Done changing to histogram. As in the queues, they are already exposed, indirectly though.